tests/unit/phpunit/modules/Emails/EmailFromValidatorTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ *
+ * SugarCRM Community Edition is a customer relationship management program developed by
+ * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+ *
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
+ * Copyright (C) 2011 - 2018 SalesAgility Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License version 3 as published by the
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+use SuiteCRM\StateCheckerPHPUnitTestCaseAbstract;
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+require_once __DIR__ . '/../../../../../modules/Emails/EmailFromValidator.php';
+
+/**
+ * EmailFromValidatorTest
+ *
+ * @author gyula
+ */
+class EmailFromValidatorTest extends StateCheckerPHPUnitTestCaseAbstract
+{
+    // TODO: !@# Needs more test...
+    public function testValidateWrongFromAddr() {
+        $email = new Email();
+        $validator = new EmailFromValidator();
+        
+        // from_addr is not set
+        $email->From = 'gusta@yammee.org';
```

Comment:
I don't see any reason not to use faker for user input here.

Reply:
faker makes random data?

Reply:
If we're calling the validate email address function would it not make sense to use a different random valid email address each time?

Reply:
Then tests are failing randomly? It could leads a painfull debugging issue when trying to figure out why is it failing but it will sometimes failing sometime doesn't...
Test's dependencies should be same at each time. Never use on the fly generated random data and/or current data/time in tests.